EpiAware/src/utilities.jl

```diff
@@ -109,6 +109,44 @@ function create_discrete_pmf(dist::Distribution; Δd = 1.0, D)
     ts .|> (t -> ∫F(dist, t, Δd)) |> diff |> p -> p ./ sum(p)
 end
 
+"""
+    neg_MGF(r, w::AbstractVector)
+
+Compute the negative moment generating function (MGF) for a given rate `r` and weights `w`.
+
+# Arguments
+- `r`: The rate parameter.
+- `w`: An abstract vector of weights.
+
+# Returns
+The value of the negative MGF.
+
+"""
+function neg_MGF(r, w::AbstractVector)
+    return sum([w[i] * exp(-r * i) for i = 1:length(w)])
+end
+
+function dneg_MGF_dr(r, w::AbstractVector)
+    return -sum([w[i] * i * exp(-r * i) for i = 1:length(w)])
+end
+
+function fast_R_to_r_approx(R₀, w::Vector{T}; newton_steps = 1) where {T<:AbstractFloat}
+    mean_gen_time = dot(w, 1:length(w))
```

Comment:
This assumes that the generation time is indexed from 1 which we enforce as an input? If so we need to say that somewhere.

Reply:
Yes we need to make that clear somewhere.